.github/workflows/build-and-test.yml

```diff
@@ -3,13 +3,15 @@ name: Build & Test
 on:
   push:
   pull_request:
+    types: [opened, synchronize]
 
 jobs:
   build-and-test:
     name: Build & Test
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```

Comment:
Revert this. We don't want a dynamic environment or else our CI can randomly crash and it'll add complexity to debug.